### PR TITLE
Add Array.map simplifications

### PR DESCRIPTION
**Pull request by jfmengels**

### CODE REVIEW

File: src/Simplify.elm [hunk 1]
```diff
@@ -6981,6 +6991,64 @@ stringDetermineLength expression =
...
+                Just fromListCall ->
+                    case AstHelpers.getListLiteral fromListCall.firstArg of
+                        Just [] ->
+                            Just (Exactly 0)
+
+                        Just (_ :: []) ->
+                            Just (Exactly 1)
+
+                        Just (el0 :: el1 :: el2Up) ->
+                            case traverse getComparableExpression (el0 :: el1 :: el2Up) of
```
Comment (lue-bird) [inline src/Simplify.elm:7036]: I think you copied this from set or somewhere. comparable etc certainly doesn't apply here.
I'd just use `listDetermineLength` for the list from `Array.fromList`

**Review by lue-bird — CHANGES_REQUESTED** [review]

File: tests/SimplifyTest.elm [hunk 2]
```diff
@@ -13944,6 +13945,193 @@ a = identity
...
+                            { message = "Array.map with an identity function will always return the same given array"
+                            , details = [ "You can replace this call by the array itself." ]
+                            , under = "Array.map"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = x
+"""
+                        ]
+        , test "should replace x |> Array.map identity by x" <|
```
Comment (lue-bird) [inline tests/SimplifyTest.elm:14059]: For all those `x` variables I'd prefer `array` but it doesn't matter much

Comment (jfmengels) [reply tests/SimplifyTest.elm]: (Let me know if you'd prefer me avoiding to force-push in the future)

File: src/Simplify.elm (hunk 1 above)
Comment (jfmengels) [reply src/Simplify.elm]: :+1:

**Review by lue-bird — APPROVED** [review]

**Pull request merged**